### ISSUE TITLE
No Android icon generated if /build/ is part of app path

### ISSUE THREAD

**Issue by ath0mas** [dwmkerr/app-icon#109]
no Android Manifest is found if /build/ is part of searchRoot, as the build folder rex excludes everything in this case
see https://github.com/dwmkerr/app-icon/blob/master/src/android/find-android-manifests.js

(this could be a new test case for https://github.com/dwmkerr/app-icon/blob/master/src/android/find-android-manifests.specs.js)

**Comment by dwmkerr** [comment]
Hi @ath0mas can you let me know the path in your example you'd like to find? I think in general it seems that excluding the `build` intermediary folders seems to be useful, I'm trying to think of a way to keep this behaviour but still fit your use-case! It might be that if search paths are specified, we exclude nothing. This is maybe a little odd but might be the easiest way.

**Comment by ath0mas** [comment]
The cordova application I create is located in `/c/workspace/cdvapps/cdvappexample/build/dev`
then I call `appIcon.generate()` from here without _searchRoot_, equivalent to running it from anywhere else with _searchRoot_ set to this same path

Detection for Android (and possibly for iOS too?) should not match on the searchRoot part of files found

**Comment by dwmkerr** [comment]
Oh I see - actually then the solution should be simple, if the project is rooted at:

```
root/folder/another/myproject
```

then the skipping of folders with 'build' in the name should *not* include the actual root folder of the project, this would solve your issue but also mean that things like:

```
root/folder/another/myproject/android/app/build
```

Would still be successfully ignored!

**Comment by dwmkerr** [comment]
@ath0mas Can you try using the branch `feat/better-android-build-folder-checking` and see if it resolves your issue? If so I will merge the change into master!

**Comment by ath0mas** [comment]
I'll try using your branch today

**Comment by ath0mas** [comment]
It works fine with this branch changes 👍 

---
To better understand: why AndroidManifest.xml detection can find multiple results?
I create a new Cordova project, add platform android and add plugins ; then running `generate` shows:
```
Found Android Manifest: platforms\android\CordovaLib\AndroidManifest.xml...
Found Android Manifest: platforms\android\app\src\main\AndroidManifest.xml...
Found Android Manifest: plugins\cordova-plugin-ABC\android\common\src\main\AndroidManifest.xml...
Found Android Manifest: plugins\cordova-plugin-ABC\android\CDVAbc\src\main\AndroidManifest.xml...
. . .
```
To me, here, the only expected result should be the 2nd: `platforms\android\app\src\main\AndroidManifest.xml...` ?

**Comment by dwmkerr** [comment]
OK! I'm going to merge in the changes and release. For the manifest multiple results, I think it is because the plugins and cordova dependencies themselves _also_ have manifests, and it is difficult to consistently and programatically work out which one is the right one. This is where you'd use `--search-path` to limit to the folder you know works!

**Comment by dwmkerr** [comment]
OK! I'm going to merge in the changes and release. For the manifest multiple results, I think it is because the plugins and cordova dependencies themselves _also_ have manifests, and it is difficult to consistently and programatically work out which one is the right one. This is where you'd use `--search-path` to limit to the folder you know works!

Published as `0.13.0`

**Comment by ath0mas** [comment]
Thx @dwmkerr for the release!

I tried using `--search-path` targeting the **platforms** dir for my cordova project so that it can work well whether I add platform android or ios. For android it is better as there is no match in _plugins_, but still get `platforms\android\CordovaLib\AndroidManifest.xml` result.

Would it be possible to exclude **CordovaLib** like done for node modules and build?
`if (file.match(new RegExp('CordovaLib'))) {return false;}`
(or to be able to pass our own custom matchers to android and ios detections?)
